pyccel/commands/pyccel_test.py

```diff
@@ -0,0 +1,201 @@
+    run in parallel are run. The function returns the return
+    code of the pytest command.
+
+    If the user stops the tests with Ctrl+C, the function will
+    print a message and exit gracefully.
+
+    This function does not provide default values for its
+    parameters, because these have to be handled by the command
+    line parser.
+
+    Parameters
+    ----------
+    folder : pathlib.Path or None
+        The local folder where the tests are located. If None, the
+        function will first try to extract the tests from a local editable
+        installation, then download them from the official GitHub repository.
+    dry_run : bool
+        If True, the function will not run the tests, but will
+        print the commands that would be run.
+    verbose : int
+        The verbosity level of the output. The higher the number,
+        the more detailed the output will be.
+    """
+
+    assert isinstance(folder, (pathlib.Path, type(None)))
+    assert isinstance(dry_run, bool)
+    assert isinstance(verbose, int)
+
+    import sys
+
+    # Pyccel must be installed
+    try:
+        import pyccel
+    except ImportError:
+        print('ERROR: It appears that Pyccel is not installed.')
+        print(install_msg)
+        sys.exit(1)
+
+    # Pytest must be installed
+    try:
+        import pytest
+    except ImportError:
+        print('ERROR: It appears that pytest is not installed.')
+        print(install_msg)
+        sys.exit(1)
+
+    # If a folder is provided, use it as the test directory
+    if folder is not None:
+        test_dir = folder.resolve()
+        if not test_dir.is_dir():
+            print(f"ERROR: The provided folder '{test_dir}' does not exist or is not a directory.")
+            sys.exit(1)
+        print(f"Using the provided folder as the test directory: {test_dir}")
+    else:
+        test_dir = None
+
+    if test_dir is None:
+        # Determine the version of Pyccel that we are using
+        direct_url = Distribution.from_name("pyccel").read_text("direct_url.json")
+#        print(f"Using Pyccel version {pyccel.__version__} from {pyccel.__file__}")
+#        print(f"Direct URL: {direct_url}")
+
+        # If a direct URL is provided, use it to determine the test directory
+        # Otherwise, download the test files from GitHub
+        if direct_url:
+            url = json.loads(direct_url)["url"]
+            if url.startswith('file://'):
+                test_dir = url.removeprefix('file://')
+                print(f"Using the local test directory from direct URL: {test_dir}")
+
+    if test_dir is None:
+        version = pyccel.__version__
+
+        # Download the test files
+        from urllib.request import urlopen
+        print(f"Downloading the test files from GitHub for pyccel-v{version}...")
+        zip_url  = f'https://github.com/pyccel/pyccel/archive/refs/tags/v{version}.zip'
+        zip_path = 'pyccel.zip'
+        with urlopen(zip_url, timeout=5) as response:
```

Comment:
Is there a reason why you moved back to using `urllib`? This library is being flagged by codacy because it is considered a security risk (by bandit : https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b310-urllib-urlopen). This is why I moved to `requests`. Maybe there is a way to whitelist this line?

Reply:
Simply because `urllib.request` is part of the Python Standard Library. `requests` is not part of our dependencies, and I would not use it for such a simple operation. My original code was using an old legacy function, the new one should be OK. Maybe I am missing some security check though.

Reply:
Given that we provide the url, I don't think there is a missing security check. Bandit has simply blacklisted this function completely

Reply:
```suggestion
        # Download the test files
        from urllib.request import urlopen
        print(f"Downloading the test files from GitHub for pyccel-v{version}...")
        zip_url  = f'https://github.com/pyccel/pyccel/archive/refs/tags/v{version}.zip'
        zip_path = 'pyccel.zip'
        with urlopen(zip_url, timeout=5) as response: # nosec urllib_urlopen
```

https://bandit.readthedocs.io/en/latest/config.html#exclusions